.github/workflows/fork-sync.yml

```
@@ -0,0 +1,40 @@
+on:
+  schedule:
+    - cron:  '0 7 * * 1,4'
+  workflow_dispatch:
+jobs:
+  sync_with_upstream:
+    runs-on: ubuntu-latest
+    name: Sync main with upstream latest
+    steps:
+    - name: Checkout stirling-labs branch
+      uses: actions/checkout@v2
+      with:
+        ref: stirling-labs
+    - name: Pull upstream changes
+      id: sync
+      uses: aormsby/Fork-Sync-With-Upstream-action@v2.3
+      with:
+        upstream_repository: XKCP/XKCP
+        upstream_branch: master
+        target_branch: stirling-labs
+        git_pull_args: '--unshallow --allow-unrelated-histories'
```

```suggestion
        git_pull_args: '--unshallow'
```
I don't think we want to merge unrelated histories... They should be linear to merge with respect to history.